src/ProgressOnderwijsUtils/SingleSignOn/SsoProcessor.cs

```diff
@@ -105,16 +105,16 @@ static NameValueCollection CreateQueryString(AuthnRequest req, [CanBeNull] strin
             if (!string.IsNullOrWhiteSpace(relayState)) {
                 result.Add("RelayState", relayState);
             }
-            result.Add("SigAlg", "http://www.w3.org/2000/09/xmldsig#rsa-sha1");
-            result.Add("Signature", Signature(result, cer.PrivateKey));
+            result.Add("SigAlg", "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256");
+            result.Add("Signature", Signature(result, cer.GetRSAPrivateKey()));
             return result;
         }
 
         [NotNull]
-        static string Signature([NotNull] NameValueCollection qs, AsymmetricAlgorithm key)
+        static string Signature([NotNull] NameValueCollection qs, [NotNull] RSA key)
         {
             var data = Encoding.UTF8.GetBytes(ToQueryString(qs));
-            var result = ((RSA)key).SignData(data, HashAlgorithmName.SHA1, RSASignaturePadding.Pkcs1);
+            var result = key.SignData(data, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);

```

Comment:
Random trivia: SHA512 is faster; and there's even a  SHA512-256

Reply:
But those are not documented as supported by surfconext: https://wiki.surfnet.nl/display/surfconextdev/Authentication+request%3A+bindings%2C+signing+and+options